src/common/imageio_rawspeed.cc

```diff
@@ -237,21 +237,22 @@ dt_imageio_retval_t dt_imageio_open_rawspeed(dt_image_t *img, const char *filena
       dt_imageio_retval_t ret = dt_imageio_open_rawspeed_sraw(img, r, mbuf);
       return ret;
     }
+    const int cpp = r->getCpp();
 
+    // checking for corrupted info supports cpp to be either 1 or 2
     if((r->getDataType() != TYPE_USHORT16) && (r->getDataType() != TYPE_FLOAT32)) return DT_IMAGEIO_FILE_CORRUPTED;
 
-    if((r->getBpp() != sizeof(uint16_t)) && (r->getBpp() != sizeof(float))) return DT_IMAGEIO_FILE_CORRUPTED;
+    if(((r->getBpp() / cpp) != sizeof(uint16_t)) && (r->getBpp() != sizeof(float))) return DT_IMAGEIO_FILE_CORRUPTED;
 
-    if((r->getDataType() == TYPE_USHORT16) && (r->getBpp() != sizeof(uint16_t))) return DT_IMAGEIO_FILE_CORRUPTED;
+    if((r->getDataType() == TYPE_USHORT16) && ((r->getBpp() / cpp) != sizeof(uint16_t))) return DT_IMAGEIO_FILE_CORRUPTED;
 
     if((r->getDataType() == TYPE_FLOAT32) && (r->getBpp() != sizeof(float))) return DT_IMAGEIO_FILE_CORRUPTED;
 
-    const float cpp = r->getCpp();
-    if(cpp != 1) return DT_IMAGEIO_FILE_CORRUPTED;
+    if((cpp < 1) || (cpp>2)) return DT_IMAGEIO_FILE_CORRUPTED;
 
     img->buf_dsc.channels = 1;
 
-    switch(r->getBpp())
+    switch(r->getBpp() / cpp)
```

Comment:
This also accepts `2*sizeof(float)`. (I.e. it assumes it would have been discarded above)
I think you want
```
    switch(r->getBpp() / cpp)
    {
      case sizeof(uint16_t):
        img->buf_dsc.datatype = TYPE_UINT16;
        break;
      case 2 * sizeof(uint16_t):
        img->buf_dsc.datatype = TYPE_UINT16;
        break;
      case sizeof(float):
        img->buf_dsc.datatype = TYPE_FLOAT;
        break;
      default:
        return DT_IMAGEIO_FILE_CORRUPTED;
    }
```

Reply:
`2 * sizeof(uint16_t)` is the same as `sizeof(float)` for most compilers / os so the compiler mocks about two same test cases.

In line 249 there is already a test for >1channel floats ...

Reply:
Ah right.
Add a comment next to the switch explaining that than.

Reply:
done